gui/advfort.lua

```diff
@@ -1109,16 +1109,17 @@ function LinkBuilding(args)
         end
     end
     if lever_bld==nil then
-        if bld:getType()==df.building_type.Trap and bld:getSubtype()==df.trap_type.Lever then
+        if bld:getType()==df.building_type.Trap and bld:getSubtype()==df.trap_type.Lever or bld:getSubtype()==df.trap_type.PressurePlate then
```

Comment:
You should use parentheses here - `A and B or C` is equivalent to `(A and B) or C`, but you want `A and (B or C)`

Reply:
Oops, you're right.